test/performance/tests/test_performance.py

```diff
@@ -30,3 +31,91 @@ def test_load_test(instances: List[harness.Instance]):
     assert "control-plane" in util.get_local_node_status(cluster_node)
     assert "control-plane" in util.get_local_node_status(joining_node)
     assert "control-plane" in util.get_local_node_status(joining_node_2)
+
+    configure_kube_burner(cluster_node)
+    process_dict = collect_metrics(instances)
+    run_kube_burner(cluster_node)
+    stop_metrics(instances, process_dict)
+    pull_metrics(instances)
+
+
+def test_single_node_load(session_instance: harness.Instance):
+    """Test the performance of a single node cluster with all features enabled."""
+    configure_kube_burner(session_instance)
+    process_dict = collect_metrics([session_instance])
+    run_kube_burner(session_instance)
+    stop_metrics([session_instance], process_dict)
+    pull_metrics([session_instance])
+
+
+def stop_metrics(instances: List[harness.Instance], process_dict: dict):
+    """Stops collecting metrics in the background from each instance."""
+    for instance in instances:
+        process_dict[instance.id].kill()
+
+
+def collect_metrics(instances: List[harness.Instance]):
+    """
+    Starts collecting metrics in the background from each instance. Returns a dictionary
+    with the process object for each instance.
+    """
+    process_dict = {}
+    for instance in instances:
+        pid = instance.exec(
+            ["pgrep", "k8s-dqlite"], text=True, capture_output=True
+        ).stdout.strip()
+        instance.exec(["apt-get", "install", "-y", "sysstat"])
+        subprocess = instance.exec_with_popen(
+            [
+                "pidstat",
+                "-druh",
+                "-p",
+                pid,
+                "1",
+                ">",
+                f"/root/{instance.id}_metrics.log",
+            ]
+        )
+        process_dict[instance.id] = subprocess
+    return process_dict
+
+
+def pull_metrics(instances: List[harness.Instance]):
+    """Pulls metrics file from each instance to the local machine."""
+    for instance in instances:
+        instance.pull_file(
+            f"/root/{instance.id}_metrics.log", f"./{instance.id}_metrics.log"
+        )
+
+
+def configure_kube_burner(instance: harness.Instance):
+    """Downloads and sets up `kube-burner` on each instance if it's not already present."""
+    if (
+        not instance.exec(["test", "-f", "/root/kube-burner"], check=False).returncode
+        == 0
+    ):
+        url = "https://github.com/kube-burner/kube-burner/releases/download/v1.2/kube-burner-1.2-Linux-x86_64.tar.gz"
```

Comment:
Maybe we can make the `kube-burner` version configurable?

Reply:
Yes, I would like this to be arch configurable too. Unfortunately, our friends at kube-burner have been inconsistent with their naming across releases, meaning we can only make the entire url configurable instead.